### PR TITLE
check NewAddDocumentCtx return value in rs_api

### LINKED ISSUE #1290: Crash when calling RediSearch_IndexAddDocument and NewAddDocumentCtx fails
When calling the API function `RediSearch_IndexAddDocument` there's a crash in case adding the document fails. This happens because `NewAddDocumentCtx` might return `NULL`. See here: https://github.com/RediSearch/RediSearch/blob/dd6b1e75ce039d923a0cae7554eaee48ece743a4/src/redisearch_api.c#L202-L204

You can recreate this by calling `RediSearch_IndexAddDocument` with a duplicate field in the index spec. Then internally we fail on ``"Tried to insert `%s` twice"`` and `NewAddDocumentCtx` returns `NULL`.

### PR DESCRIPTION
**Pull request by ashtul**
fix #1290

### CODE REVIEW

**Comment by ashtul** [comment]
@MeirShpilraien 
This fails when same field is added twice.
cpp test added.

**Comment by MeirShpilraien** [comment]
@ashtul notice that the tests are still failing

**Comment by ashtul** [comment]
Yes. Didn’t figure out yet why it won’t clean like in other tests.

File: src/cpptests/t_llapi.cpp [hunk 1]
```diff
@@ -727,3 +727,22 @@ TEST_F(LLApiTest, testFreeDocument) {
...
+  // adding text field to the index
+  RediSearch_CreateField(index, FIELD_NAME_1, RSFLDTYPE_FULLTEXT, RSFLDOPT_NONE);
+
+  // adding document to the index
+  Document* d = RediSearch_CreateDocumentSimple("doc1");
+
+  // adding same field twice
+  RediSearch_DocumentAddFieldCString(d, FIELD_NAME_1, "some test to field", RSFLDTYPE_DEFAULT);
+  RediSearch_DocumentAddFieldCString(d, FIELD_NAME_1, "some test to same field", RSFLDTYPE_DEFAULT);
+  RediSearch_SpecAddDocument(index, d);
```
Comment (MeirShpilraien) [inline src/cpptests/t_llapi.cpp:743]: Why not assert that this returns REDISMODULE_ERR?

Comment (ashtul) [reply src/cpptests/t_llapi.cpp]: done

**Pull request merged**